nms/scripts/generateAPIFromSwaggerForTypeScript.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+################################################################################
+# Copyright 2022 The Magma Authors.
+
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+set -e
+
+USAGE="generateAPIFromSwaggerForTypeScript.sh
+— generates TypeScript NMS API bindings for swagger spec
+
+  Usage:
+    generateAPIFromSwaggerForTypeScript.sh <input> <output>
+
+  Options:
+    <input>   Input swagger.yml file to read.
+    <input>   Output file for js bindings.
+"
+
+if [ $# -ne 2 ]; then
+  echo "$USAGE"
+  exit 1
+fi
+
+INPUT=${1}
+OUTPUT=${2}
+echo "Input file: $INPUT";
+echo "Output directory: $OUTPUT";
+
+yarn --silent openapi -i "${INPUT}" -o "${OUTPUT}"
+
+HEADER='/**
+ * Copyright 2022 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @generated
+ * This file is generated by "nms/scripts/generateAPIFromSwaggerForTypeScript.sh".
+ */
+'
+
+# shellcheck disable=SC2044
+for file in $(find "${OUTPUT}" -type f); do
```

Comment:
If someone accidentally uses an `$OUTPUT` folder that contains more files, this might break a lot. Please add a more restrictive way of finding the generated files.